clean_old_aggregates.py

```diff
@@ -0,0 +1,70 @@
+import keystoneauth1
+import datetime
+from keystoneauth1.identity import v3
+from keystoneauth1 import loading
+from keystoneauth1 import session
+from novaclient import client as nova_client
+from blazarclient import client as blazar_client
+
+def main():
+
+    loader = loading.get_plugin_loader('password')
+    auth = loader.load_from_options(auth_url='http://REDACTED:35357',
+    username='admin',
+    password='REDACTED',
+    project_id='REDACTED',
+    user_domain_id ='default')
+
+    sess = session.Session(auth=auth)
+
+    nova = nova_client.Client(version='2', region_name='CHI@TACC', session=sess)
+    blazar = blazar_client.Client(1, session=sess, service_type='reservation', region_name='CHI@TACC')
+
+    term_leases=[]
+    agglist=[]
+    aggdetails=[]
+    aggnames=[]
+    exp_aggs=[]
+    agg_host=[]
+
+    # Find leases with end date before current time
+    leases=blazar.lease.list()
+
+    time=datetime.datetime.now()
+    for x in leases:
+        lease_time=datetime.datetime.strptime(x['end_date'], '%Y-%m-%dT%H:%M:%S.%f')
+        if lease_time < time:
+            print("lease ending {} has ended. Time {}".format(lease_time,time))
+            term_leases.append(x['reservations'][0]['id'])
+
+    # Find aggregates matching ended leases
+    aggregates=nova.aggregates.list()
+
+    for agg in aggregates:
+        ags=str(agg)
+        agglist.append(ags[12:17])
```

Comment:
I'm not sure what's going on here, but it looks like you're casting to a string and operating on the string representation generated when the object is printed? That can break very easily in the future; instead, you can pull out exactly what you need, or, rather, just keep the list of aggregates untouched!

```
# (Using auth method from earlier in review)
aggregates = osrest.nova.aggregates(auth)
```

Keeping all of these parallel arrays makes things harder for us. I imagine that all the info we need is here in the aggregate list (names and IDs I would expect would be here.)